src/providers/realtime-provider.js

```diff
@@ -162,44 +198,77 @@ export default class RealtimeProvider {
         const domainObject = subscriptionDetails.domainObject;
         const message = MESSAGES.SUBSCRIBE[domainObject.type](subscriptionDetails);
 
-        this.sendOrQueueMessage(message);
+        this.sendMessage(message);
     }
 
     sendUnsubscribeMessage(subscriptionDetails) {
         let message = MESSAGES.UNSUBSCRIBE(subscriptionDetails);
 
-        this.sendOrQueueMessage(message);
+        this.sendMessage(message);
     }
 
-    reconnect() {
-        this.subscriptionsByCall.clear();
+    #setCallFromClock(clock) {
+        const correspondingSubscription = Object.values(this.subscriptionsById).find(subscription => {
+            return subscription.domainObject.identifier.key === clock.identifier.key;
+        });
 
-        if (this.reconnectTimeout) {
-            return;
+        if (correspondingSubscription !== undefined) {
+            this.remoteClockCallNumber = correspondingSubscription.call.toString();
+        } else {
+            delete this.remoteClockCallNumber;
         }
+    }
 
-        this.reconnectTimeout = setTimeout(() => {
-            this.connect();
-            delete this.reconnectTimeout;
-        }, FALLBACK_AND_WAIT_MS[this.currentWaitIndex]);
+    #processBatchQueue(batchQueue, call) {
+        let subscriptionDetails = this.subscriptionsByCall.get(call);
+        let telemetryData = [];
 
-        if (this.currentWaitIndex < FALLBACK_AND_WAIT_MS.length - 1) {
-            this.currentWaitIndex++;
+        // possibly cancelled
+        if (!subscriptionDetails) {
+            return;
         }
-    }
 
-    sendOrQueueMessage(request) {
-        if (this.connected) {
-            try {
-                this.sendMessage(request);
-            } catch (error) {
-                this.connected = false;
-                this.requests.push(request);
-                console.error("🚨 Error while attempting to send to websocket, closing websocket", error);
-                this.socket.close();
-            }
-        } else {
-            this.requests.push(request);
+        batchQueue.forEach((rawMessage) => {
+            const message = JSON.parse(rawMessage);
```

Comment:
do we want to `catch` if the parse fails for some reason?

Reply:
There's no expected scenario in which it fails, so I'm inclined to fail fast here. Something's gone very wrong if this code receives invalid JSON.